oracle.go

```diff
@@ -65,14 +66,34 @@ func checkMongoSupportsOut(db *mgo.Database) bool {
 	return v[0] > 2 || (v[0] == 2 && v[1] >= 6)
 }
 
+// completedOldTransactionMatch creates a search parameter for transactions
+// that are flagged as completed, and were generated older than the given
+// timestamp. If the timestamp is empty,then only the completed status is evaluated.
+// The returned object is suitable for being passed to a $match or a Find() operation.
+func completedOldTransactionMatch(timestamp time.Time) bson.M {
+	var zeroTime = time.Time{}
+	if timestamp == zeroTime {
+		return bson.M{"s": bson.M{"$gte": taborted}}
+	}
+	tid := bson.NewObjectIdWithTime(timestamp)
+	return bson.M{
+		"s":   bson.M{"$gte": taborted},
+		"_id": bson.M{"$lt": tid},
+	}
+
+}
```

Comment:
Why double up the `taborted` bit? Also, `time.Time` has an `IsZero` method. I suggest this:

```go
match := bson.M{
    "s": bson.M{"$gte": taborted}
}
if !timestamp.IsZero() {
    match["_id"] = bson.M{"$lt": bson.NewObjectIdWithTime(timestamp)}
}
return match
```